### PR TITLE
Add slint parser configuration and queries

### PR DESCRIPTION
**Pull request by jrmoulton**
Is there anything else that needs to be changed to have this merged?

### CODE REVIEW

File: lua/nvim-treesitter/parsers.lua [hunk 1]
```diff
@@ -461,6 +461,17 @@ list.supercollider = {
...
 }
 
+list.slint = {
+  install_info = {
+    url = "https://github.com/jrmoulton/tree-sitter-slint",
+    files = { "src/parser.c" },
+    branch = "main",
+  },
+  maintainers = { "@jrmoulton" },
+  filetype = "slint",
```
Comment (clason) [inline lua/nvim-treesitter/parsers.lua:471]: I don't think you need that if your parser has the same name as the filetype.

**Comment by clason** [comment]
You could also add the parser (with the matching revision) to the `lockfile.json` -- this way users don't have to wait for the automated update (which is currently blocked by another parser).

**Comment by jrmoulton** [comment]
Those should be done now

File: queries/slint/highlights.scm [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+(comment) @comment
+
+(value) @number
+(string) @string
+
+[
+"struct"
+"property"
+"callback"
+"import"
```
Comment (theHamsta) [inline queries/slint/highlights.scm:25]: `@include`?

File: queries/slint/highlights.scm [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+"from"
+"root"
+"parent"
+"this"
+"for"
```
Comment (theHamsta) [inline queries/slint/highlights.scm:30]: `@conditional` `@repeat`?

Comment (jrmoulton) [reply queries/slint/highlights.scm]: Thanks for the reviews. I appreciate it

File: queries/slint/highlights.scm [hunk 4]
```diff
@@ -18,19 +18,29 @@
...
+
 [
 "struct"
 "property"
 "callback"
-"import"
-"from"
 "root"
 "parent"
 "this"
```
Comment (theHamsta) [inline queries/slint/highlights.scm:42]: Is `this` a variable.builtin/constant.builtin? I don't know what the semantics of `in` is
 In some languages it is a keyword.operator

Comment (jrmoulton) [reply queries/slint/highlights.scm:42]: Oh you're right, they should be variable.builtin. And after looking at other languages queries it seems like `in` should be `@repeat`

**Comment by jrmoulton** [comment]
If there are any other changes that should be made to the highlights I can get them done

**Comment by jrmoulton** [comment]
Perfect. Thanks for all the reviews!

File: queries/slint/highlights.scm [hunk 5]
```diff
@@ -11,7 +11,7 @@
 (function_identifier) @function
 
 (reference_identifier) @keyword
-(visibility_modifier) @keyword
+(visibility_modifier) @include
```
Comment (theHamsta) [inline queries/slint/highlights.scm:14]: I assume that `visibility_modifier` is something like public/private. This is usually not used in such a way in other languages. Or is the visibility_modifier directly used on import?

Comment (jrmoulton) [reply queries/slint/highlights.scm:14]: Gotcha. For now it just references `export` but it's used in a similar way to public/private. In that case I'll change it back

Comment (theHamsta) [reply queries/slint/highlights.scm]: Ok, then you were probably right. `export` should be `@include`. Sorry for the confusion.

**Review by theHamsta — APPROVED** [review]

**Comment by jrmoulton** [comment]
Thanks!

**Pull request merged**